src/charts/stacked-bar.js

```diff
@@ -473,44 +471,42 @@ define(function(require){
 
         /**
          * Draws the bars along the y axis
-         * @param  {D3Selection} bars Selection of bars
+         * @param  {D3Selection} layersSelection Selection of bars
          * @return {void}
          */
-        function drawVerticalBars(series) {
-            // Enter + Update
-            let bars = series
-                .data(layers)
+        function drawVerticalBars(layersSelection) {
+            let layerJoin = layersSelection
+                .data(layers);
+
+            layerElements = layerJoin
                 .enter()
                   .append('g')
-                    .classed('layer', true)
                     .attr('fill', (({key}) => categoryColorMap[key]))
+                    .classed('layer', true);
+
+            let barJoin = layerElements
                     .selectAll('.bar')
-                    .data((d) => d)
+                    .data((d) => d);
+
+            // Enter + Update
+            let bars = barJoin
                     .enter()
                       .append('rect')
                         .classed('bar', true)
                         .attr('x', (d) => xScale(d.data.key))
                         .attr('y', (d) => yScale(d[1]))
                         .attr('width', xScale.bandwidth )
-                        .attr('fill', (({data}) => categoryColorMap[data.stack+data.key])),context;
+                        .attr('fill', (({data}) => categoryColorMap[data.stack+data.key]));
```

Comment:
could we do string interpolation here? `${data.stack}${data.key}`, for a second i thought you might be adding numbers

Reply:
Totally!